src/Sylius/Bundle/AdminBundle/Controller/Dashboard/StatisticsController.php

```diff
@@ -21,20 +21,11 @@
 
 final class StatisticsController
 {
-    /** @var EngineInterface|Environment */
-    private $templatingEngine;
-
-    private StatisticsDataProviderInterface $statisticsDataProvider;
-
     /**
      * @param EngineInterface|Environment $templatingEngine
      */
-    public function __construct(
-        object $templatingEngine,
-        StatisticsDataProviderInterface $statisticsDataProvider
-    ) {
-        $this->templatingEngine = $templatingEngine;
-        $this->statisticsDataProvider = $statisticsDataProvider;
+    public function __construct(private object $templatingEngine, private StatisticsDataProviderInterface $statisticsDataProvider)
```

Comment:
```suggestion
    public function __construct(private EngineInterface|Environment $templatingEngine, private StatisticsDataProviderInterface $statisticsDataProvider)
```

But maybe this is a BC-break.

Reply:
It is BC-Break :( 

On the other hand, the code won't work if anything else would be injected (unless someone duplicate the contract under different name). Also, we had "soft" requirement for these classes. I would introduce it anyway

Reply:
👍